### PR TITLE
Wip/drothe image support

### PR DESCRIPTION
**Pull request by d-rothe**
# 4 pull requests belonging together (same branch name): DeviceAccess, ControlSystemAdapter, DoocsAdapter, DoocsBackend.

### CODE REVIEW

**Comment by mhier** [comment]
Attention: commit message pitfalls ahead :-) The entire PR seems to be intended as a single commit (squashing), but the PR title will then be the commit message. Commit messages should never contain "WIP" (in master branch) and should ideally always follow conventional commits. I recommend preparing the right commit structure and message in the branch already (you can squash all commits easily with an interactive rebase) and then merge without squashing. Otherwise there is a high risk that Github does something unexpected and the result in the master is not as desired.

File: include/DoocsBackendImageRegisterAccessor.h [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+#include <ChimeraTK/MappedImage.h>
+
+namespace ChimeraTK {
+
+  /// implements conversion from DOOCS image -> MappedImage
+  struct MappedDoocsImgIn : public MappedImage {
+    using MappedImage::MappedImage;
+
+    ImgFormat getImgFormat(const IMH* h) {
```
Comment (mhier) [inline include/DoocsBackendImageRegisterAccessor.h:18]: Please move implementation to .cc file. (If not possible e.g. for templates, move them at least below the class declaration, like the inline functions below - btw: inlining is no performance here, so please move them to the .cc file as well)

File: include/DoocsBackendImageRegisterAccessor.h [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      hOut->x_start = h->x_start;
+      hOut->y_start = h->y_start;
+      hOut->scale_x = h->scale_x;
+      hOut->scale_y = h->scale_y;
+
+      memcpy(imgBody(), body, useLines * bytesPerLine);
+    }
+  };
+
+  class DoocsBackendImageRegisterAccessor : public DoocsBackendRegisterAccessor<std::uint8_t> {
```
Comment (mhier) [inline include/DoocsBackendImageRegisterAccessor.h:66]: Please add missing chrome bar `/******/`

File: include/DoocsBackendImageRegisterAccessor.h [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+#include "DoocsBackendRegisterAccessor.h"
+#include <eq_client.h>
+#include <type_traits>
+
+#include <ChimeraTK/MappedImage.h>
+
+namespace ChimeraTK {
+
+  /// implements conversion from DOOCS image -> MappedImage
+  struct MappedDoocsImgIn : public MappedImage {
```
Comment (mhier) [inline include/DoocsBackendImageRegisterAccessor.h:15]: Please swap the two class/struct definitions, so the more important one comes first.

File: src/DoocsBackend.cc [hunk 4]
```diff
@@ -5,27 +5,32 @@
...
+#include "DoocsBackendImageRegisterAccessor.h"
 #include "DoocsBackendIntRegisterAccessor.h"
 #include "DoocsBackendLongRegisterAccessor.h"
 #include "DoocsBackendStringRegisterAccessor.h"
-#include "DoocsBackendEventIdAccessor.h"
 #include "DoocsBackendTimeStampAccessor.h"
 #include "RegisterInfo.h"
 #include "StringUtility.h"
 #include "ZMQSubscriptionManager.h"
+#include <libxml++/libxml++.h>
```
Comment (mhier) [inline src/DoocsBackend.cc:24]: Do we really need all of these new headers?

**Review by mhier — APPROVED** [review]

File: include/DoocsBackendImageRegisterAccessor.h (hunk 1 above)
Comment (d-rothe) [reply include/DoocsBackendImageRegisterAccessor.h]: done

File: src/DoocsBackend.cc (hunk 4 above)
Comment (d-rothe) [reply src/DoocsBackend.cc:24]: I now removed libxml header

**Pull request merged**